core/src/main/java/org/apache/iceberg/SnapshotManager.java

```diff
@@ -19,292 +19,48 @@
 
 package org.apache.iceberg;
 
-import java.util.List;
-import java.util.Map;
-import org.apache.iceberg.exceptions.CherrypickAncestorCommitException;
-import org.apache.iceberg.exceptions.ValidationException;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
-import org.apache.iceberg.util.PartitionSet;
-import org.apache.iceberg.util.PropertyUtil;
-import org.apache.iceberg.util.SnapshotUtil;
-import org.apache.iceberg.util.WapUtil;
 
-public class SnapshotManager extends MergingSnapshotProducer<ManageSnapshots> implements ManageSnapshots {
+public class SnapshotManager implements ManageSnapshots {
```

Comment:
These changes are safe because the superclass was never exposed. The `Table` API returns `ManageSnapshots` and this still implements that API.

Reply:
With this change, `ManageSnapshots` is basically a wrapper of running transactions for snapshot-related operations. Do you think we will eventually add it as a part of the `Transaction`, so people can do

```java
Transaction transaction = table.newTransaction();
transaction.manageSnapshots()
  .createBranch("dev")
  .cherrypick("dev", 123L)
  .commit();
transaction.newAppend().appendFiles(f1, f2).toBranch("dev").commit();
transaction.commit();
```

Or we do not add this to transactions, and we do:

```java
Transaction transaction = table.newTransaction();
transaction.updateRefs().createBranch("dev").commit();
transaction.cherryPick().cherrypick("dev", 123L).commit();
transaction.newAppend().appendFiles(f1, f2).toBranch("dev").commit();
transaction.commit();
```

Which means we will still need a new API for ref updates.

Do you see any benefits for any of the approach?

Reply:
I think we _could_ implement this as part of a transaction by giving the `Transaction` a way to pass itself into the API. But for now, I think it's reasonable to not mix the two. `Transaction` will currently fail if you attempt to call `manageSnapshots()` on it.

Your second idea is also interesting and maybe a better way of combining a generic transaction with these operations. Let's wait until there's a use case for it though.